nemo/collections/asr/models/ctc_models.py

```diff
@@ -99,8 +99,30 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         super().__init__(cfg=cfg, trainer=trainer)
         self.preprocessor = EncDecCTCModel.from_config_dict(self._cfg.preprocessor)
         self.encoder = EncDecCTCModel.from_config_dict(self._cfg.encoder)
+
+        if "params" in self._cfg.decoder:
+            if "feat_in" not in self._cfg.decoder.params or (
+                not self._cfg.decoder.params.feat_in and hasattr(self.encoder, '_feat_out')
+            ):
+                self._cfg.decoder.params.feat_in = self.encoder._feat_out
+            if "feat_in" not in self._cfg.decoder.params or not self._cfg.decoder.params.feat_in:
+                raise ValueError("param feat_in of the decoder's config is not set!")
+        else:
+            if "feat_in" not in self._cfg.decoder or (
+                not self._cfg.decoder.feat_in and hasattr(self.encoder, '_feat_out')
+            ):
+                self._cfg.decoder.feat_in = self.encoder._feat_out
+            if "feat_in" not in self._cfg.decoder or not self._cfg.decoder.feat_in:
+                raise ValueError("param feat_in of the decoder's config is not set!")
```

Comment:
Does this work without OmegaConf's open_dict() or is this not an OmegaConf config?